aiidalab_widgets_base/structures.py

```diff
@@ -394,19 +404,51 @@ def _validate_and_fix_ase_cell(self, ase_structure, vacuum_ang=10.0):
     def _on_file_upload(self, change=None):
         """When file upload button is pressed."""
         for fname, item in change["new"].items():
-            frmt = fname.split(".")[-1]
-            if frmt == "cif":
-                self.structure = CifData(file=io.BytesIO(item["content"]))
-            else:
-                with tempfile.NamedTemporaryFile(suffix=f".{frmt}") as temp_file:
-                    temp_file.write(item["content"])
-                    temp_file.flush()
-                    self.structure = self._validate_and_fix_ase_cell(
-                        get_ase_from_file(temp_file.name)
-                    )
+            self.structure = self._read_structure(fname, item["content"])
             self.file_upload.value.clear()
             break
 
+    def _read_structure(self, fname, content):
+        suffix = "".join(pathlib.Path(fname).suffixes)
+        if suffix == ".cif":
+            from CifFile import StarError
+
+            try:
+                return CifData(file=io.BytesIO(content))
+            except StarError:
+                self._status_message.message = f"Could not parse CIF file {fname}"
+                return None
+
+        with tempfile.NamedTemporaryFile(suffix=suffix) as temp_file:
+            temp_file.write(content)
+            temp_file.flush()
+            try:
+                structures = get_ase_from_file(temp_file.name)
+            except ValueError as e:
+                self._status_message.message = f"""
+                    <div class="alert alert-info">
+                    <strong>ERROR: {e}</strong>
+                    </div>
+                    """
```

Comment:
```suggestion
                self._status_message.message = f"""
                    <div class="alert alert-danger">ERROR: {e}</div>
                    """
```

I would suggest making it alert-danger and don't use bold text. It is already bold enough :)

Same for the occurrences below.